### PR TITLE
Fixes found during integration

### PR DESCRIPTION
**Pull request by kamil-kielbasa-aa**
- additional protection for VLA buffer check against unsigned integer overflow.
- for `responder.c` preprocessor condition for windows compiler msvc which does not support VLA at all.
- fixes for oscore found on internal tests:
  - wrong buffer size with enabled VLA.
  - lazy check to defer memcmp.

### CODE REVIEW

File: src/edhoc/responder.c
Comment (StefanHri) [inline src/edhoc/responder.c:1]: The MAC size is defined by the suite.

File: src/oscore/coap2oscore.c
Comment (StefanHri) [inline src/oscore/coap2oscore.c:1]: Why is this necessary? By doing this you lose potential benefit from VLA.

**Comment by StefanHri** [comment]
Please answer or provide a fix for the comments above.

File: src/edhoc/responder.c
Comment (kamil-kielbasa-aa) [reply src/edhoc/responder.c:1]: You are right. But internally we are compiling uEDHOC for windows, linux and mac. Windows and build-in compiler msvc does not support VLA at all (also in latest [C11, C17](https://devblogs.microsoft.com/cppblog/c11-and-c17-standard-support-arriving-in-msvc/) standards).

Because `get_aead_mac_len(rc->suite.edhoc_aead)` is known at run-time we will not be able to compile uEDHOC for Windows and it does not matter whether we turn on or off `VLA` flag.

This is why I put max aead mac length value. If you have better idea how to fix above problem, please share.
